### PR TITLE
update e2e-framework test infra to go1.18

### PR DESCRIPTION
**Pull request by harshanarayana**
- needed by https://github.com/kubernetes-sigs/e2e-framework/pull/142

/assign @vladimirvivien  @ShwethaKumbla

### CODE REVIEW

**Comment by harshanarayana** [comment]
@vladimirvivien need an LGTM here as well kind sir before we can get the e2e-framework pr merged..

**Review by cpanato — APPROVED** [review]
/lgtm

**Pull request merged**
